test/acs-engine-test/metrics/metrics.go

```diff
@@ -0,0 +1,38 @@
+package metrics
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/alexcesaro/statsd"
+)
+
+type mdmBucket struct {
+	Namespace string            `json:"Namespace"`
+	Metric    string            `json:"Metric"`
+	Dims      map[string]string `json:"Dims"`
+}
+
+func AddMetric(namespace, metric string, count int64, dims map[string]string) error {
+	bucket := mdmBucket{
+		Namespace: namespace,
+		Metric:    metric,
+		Dims:      dims}
+	data, err := json.Marshal(bucket)
+	if err != nil {
+		return err
+	}
+	client, err := statsd.New(
+		statsd.Address(":8125"),
+		statsd.Network("udp"),
+		statsd.ErrorHandler(
+			func(err error) {
+				fmt.Println(err.Error())
+			}))
+	if err != nil {
+		return err
+	}
+	defer client.Close()
+	client.Count(string(data), count)
```

Comment:
We should discuss more on the design if you are planning to use this as a generic class for ACS RP. Plus, I am not sure if MDM metrics code for RP needs to live in ACS Engine. IMO that's out of scope of what ACS Engine should do.

Also please check the (old) ACS RP code to see examples of how metrics are logged and used currently (CRP metrics logging also has some good examples).